server/service/store/handler_store_data.go

```diff
@@ -0,0 +1,59 @@
+package store
+
+import (
+	"encoding/hex"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func HandleStoreData(w http.ResponseWriter, r *http.Request) {
+	dataHashStr := mux.Vars(r)["key"]
+	_, err := hex.DecodeString(dataHashStr)
+	if err != nil {
+		log.Errorf("failed to decode dataHash: %s", err.Error())
+		http.Error(w, "failed to decode dataHash", http.StatusBadRequest)
+		return
+	}
+
+	dealIDStr := mux.Vars(r)["dealId"]
+
+	body, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		log.Errorf("failed to read data: %v", err.Error())
+		http.Error(w, "failed to read data", http.StatusBadRequest)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
```

Comment:
You don't have to set status code OK here. You have a lot of logics remaining below, `os.Mkdir`, `os.OpenFile` and so on. If you get an error from those logics, you have to return a non-OK status code, such as 500.

Also, at those logics below, you just print a error log and proceed to the next step. It'll cause additional errors.

Reply:
Pls keep in mind that your service must return a successful response only if all internal operations are successfully done. In other words, you must return OK only if all data body is written to the disk successfully. If not, you don't have to return OK.